test/test_framework.py

```diff
@@ -718,6 +731,12 @@ def test_helper_properties(self):
         self.assertEqual(my_obj.meta, framework.meta)
         self.assertEqual(my_obj.charm_dir, framework.charm_dir)
 
+    def test_ban_concurrent_frameworks(self):
+        framework = self.create_framework()
+        with self.assertRaises(OperationalError):
+            framework_copy = self.create_framework()
```

Comment:
This is a case where I would actually look at the exception string instead of the type. We don't care as much about "OperationalError" as we do care about "database is locked". It's also in a test, so that conversation we had doesn't apply. If that error message changes we'll know it because we're always verifying it, and at the right time (testing, not in production).

Reply:
https://github.com/canonical/operator/pull/141/commits/f6004043ef0519edfe7dee7aace2ccab22300d4a